Cannot upload a resource and commit changes made to GitHub from a newly created dataset

When I upload a dataset, I expect it to have been uploaded once successfully so I can fill the necessary information (describe the fields + metadata) and be able to click on `Save` at the end of the flow, saving the metadata to GitHub.

## Acceptance

* [ ] I can save metadata from any uploaded resource (e.g. this one https://storage.googleapis.com/gift-datasets/gift-data/MDEwOlJlcG9zaXRvcnkzNDM4NjcwOTc%3D/8f2f06c279461768c64f11250b2615fbc8059bd743f58a2f40b4541e53244a65) successfully.

## Tasks

* [x] Validate user cookie method
* [x] Remove cookie when user log out
* [x] Update pages to validate `session` and `cookie

## Analysis

It looks like a permission issue. I was able to push a large file here https://github.com/gift-data/costa-rica-presupuesto-ingresos-2020/blob/main/data/simple_heavy4.csv for an existing repo (created by @sglavoie back in September 2020) but nothing at all to a new repo like this one (created by @sglavoie on March 2, 2021) https://github.com/gift-data/paraguay-2017

I thought maybe it had to do with the fact that I deleted the repo and created a new one with the same name, but the same happens with this other one I created later: https://github.com/gift-data/paraguay-2017-2

Simple files (4 columns) of different sizes could be used from this dataset (see in Google Cloud): gift-datasets/gift-data/MDEwOlJlcG9zaXRvcnkzNDM4NjcwOTc=

----

**Note:** Once the metadata is updated and the button `Save` is clicked, the message `Uploaded successfully.` is shown when the information was successfully saved. However, no resource should be uploaded at this point: only the metadata should be updated, so the message is misleading.

When starting from the beginning of the flow by clicking on `Create a fiscal data schema`, all steps can succeed until one clicks on the `Save` button, at which point the message `An Error occurred when uploading the dataset!` appears and nothing gets saved to GitHub. This is true when trying to upload a file for this dataset https://github.com/gift-data/paraguay-2017, the file being online here https://storage.googleapis.com/gift-datasets/gift-data/MDEwOlJlcG9zaXRvcnkzNDM4NjcwOTc%3D/8f2f06c279461768c64f11250b2615fbc8059bd743f58a2f40b4541e53244a65 (as can be seen, the dataset ID is `MDEwOlJlcG9zaXRvcnkzNDM4NjcwOTc=`).

The above upload, once the data is described and the metadata filled, shows the following error in the JavaScript console:

```
853903f82485797d365e8e24b62de66875bdb751.51264a085ddf6bb56323.js:1 POST https://gift-portal-git-master-datopian1.vercel.app/api/dataset/paraguay-2017 400
(anonymous) @ 853903f82485797d365e8e24b62de66875bdb751.51264a085ddf6bb56323.js:1
e.exports @ 853903f82485797d365e8e24b62de66875bdb751.51264a085ddf6bb56323.js:1
UnBK.e.exports @ 853903f82485797d365e8e24b62de66875bdb751.51264a085ddf6bb56323.js:1
Promise.then (async)
CgaS.c.request @ 853903f82485797d365e8e24b62de66875bdb751.51264a085ddf6bb56323.js:1
(anonymous) @ 853903f82485797d365e8e24b62de66875bdb751.51264a085ddf6bb56323.js:1
(anonymous) @ 853903f82485797d365e8e24b62de66875bdb751.51264a085ddf6bb56323.js:1
s @ 74a5bdab7c46be5812a5ed28b29adc92ff928d99.5c3b22bc5e17b1375a1c.js:1
(anonymous) @ 74a5bdab7c46be5812a5ed28b29adc92ff928d99.5c3b22bc5e17b1375a1c.js:1
(anonymous) @ 74a5bdab7c46be5812a5ed28b29adc92ff928d99.5c3b22bc5e17b1375a1c.js:1
m @ 853903f82485797d365e8e24b62de66875bdb751.51264a085ddf6bb56323.js:1
s @ 853903f82485797d365e8e24b62de66875bdb751.51264a085ddf6bb56323.js:1
(anonymous) @ 853903f82485797d365e8e24b62de66875bdb751.51264a085ddf6bb56323.js:1
(anonymous) @ 853903f82485797d365e8e24b62de66875bdb751.51264a085ddf6bb56323.js:1
onSubmit @ 853903f82485797d365e8e24b62de66875bdb751.51264a085ddf6bb56323.js:1
We @ framework.0c239260661ae1d12aa2.js:1
Ye @ framework.0c239260661ae1d12aa2.js:1
(anonymous) @ framework.0c239260661ae1d12aa2.js:1
xr @ framework.0c239260661ae1d12aa2.js:1
Cr @ framework.0c239260661ae1d12aa2.js:1
(anonymous) @ framework.0c239260661ae1d12aa2.js:1
De @ framework.0c239260661ae1d12aa2.js:1
(anonymous) @ framework.0c239260661ae1d12aa2.js:1
Or @ framework.0c239260661ae1d12aa2.js:1
Zt @ framework.0c239260661ae1d12aa2.js:1
Jt @ framework.0c239260661ae1d12aa2.js:1
t.unstable_runWithPriority @ framework.0c239260661ae1d12aa2.js:1
Wl @ framework.0c239260661ae1d12aa2.js:1
Me @ framework.0c239260661ae1d12aa2.js:1
Xt @ framework.0c239260661ae1d12aa2.js:1
853903f82485797d365e8e24b62de66875bdb751.51264a085ddf6bb56323.js:1 Error: Request failed with status code 400
    at LYNF.e.exports (853903f82485797d365e8e24b62de66875bdb751.51264a085ddf6bb56323.js:1)
    at Rn+g.e.exports (853903f82485797d365e8e24b62de66875bdb751.51264a085ddf6bb56323.js:1)
    at XMLHttpRequest.h.onreadystatechange (853903f82485797d365e8e24b62de66875bdb751.51264a085ddf6bb56323.js:1)
```

The file size doesn't matter.

--- a/src/App.js
+++ b/src/App.js
@@ -180,7 +180,7 @@ export class DatasetEditor extends React.Component {
       })
         .then((response) => {
           this.setState({ dataset: temp_dataset, resource: {} });
-          alert("Resource has been removed sucessfully");
+          alert("Resource has been removed successfully");
         })
         .catch((error) => {
           console.log(error);

--- a/src/App.js
+++ b/src/App.js
@@ -18,15 +18,6 @@ export class DatasetEditor extends React.Component {
     dataset.encoding = "utf_8";
     dataset.format = "csv";
 
-    if (
-      !("sample" in dataset) &&
-      "resources" in dataset &&
-      dataset["resources"].length > 0
-    ) {
-      dataset["sample"] = dataset["resources"][0]["sample"];
-    } else {
-      dataset["sample"] = [];
-    }
     this.state = {
       dataset,
       resource: {}, //This will hold the uploaded resource metadata
@@ -58,7 +49,7 @@ export class DatasetEditor extends React.Component {
       dataset,
       resource: updatedResource,
       tablePreviewSample: resource.tablePreviewSample,
-      tablePreviewColumns: resource.tablePreviewColumns
+      tablePreviewColumns: resource.tablePreviewColumns,
     });
   }
 
@@ -82,11 +73,6 @@ export class DatasetEditor extends React.Component {
       dataset["resources"] = [resource];
     }
     resource["sample"] = fileResource.sample;
-
-    if (dataset["sample"].length == 0) {
-      dataset["sample"] = fileResource.sample;
-    }
-
     return { dataset, resource };
   }
 
@@ -131,27 +117,26 @@ export class DatasetEditor extends React.Component {
 
   mapDatasetToFiscalFormat = (resource) => {
     const dataset = { ...this.state.dataset };
-
-    resource.schema.fields.forEach((f) => {
-      f.type = f.columnType;
-      delete f.columnType; //os-types requires type to be of rich type and will not accept the property colunmType
-    });
-    let fdp = new TypeProcessor().fieldsToModel(resource["schema"]["fields"]);
-
-    if (!Object.keys(dataset).includes("model")) {
-      dataset.model = fdp.model;
-    }
-    resource.schema.fields = Object.values(fdp.schema.fields);
-
-    dataset.resources.map((res) => {
+    const resources = dataset.resources.map((res) => {
       if (res.hash == resource.hash) {
+        resource.schema.fields.forEach((f) => {
+          f.type = f.columnType;
+          delete f.columnType; //os-types requires type to be of rich type and will not accept the property colunmType
+        });
+        let fdp = new TypeProcessor().fieldsToModel(
+          resource["schema"]["fields"]
+        );
+
+        resource.schema.fields = Object.values(fdp.schema.fields);
+        resource.model = fdp.model;
         return resource;
       } else {
         return res;
       }
     });
+
     this.setState({
-      dataset,
+      dataset: { ...dataset, resources },
     });
   };
 
@@ -165,16 +150,15 @@ export class DatasetEditor extends React.Component {
       const temp_dataset = { ...dataset };
       let path;
       if (temp_dataset.resources.length == 1) {
-        path = temp_dataset.resources[0].path
+        path = temp_dataset.resources[0].path;
         temp_dataset.resources = [];
       } else {
         const newResource = temp_dataset.resources.filter((resource) => {
-          if (resource.hash == hash){
-            path = resource.path
+          if (resource.hash == hash) {
+            path = resource.path;
           }
-          return resource.hash != hash
-        }
-        )
+          return resource.hash != hash;
+        });
         temp_dataset.resources = newResource;
       }
       axios({
@@ -184,13 +168,15 @@ export class DatasetEditor extends React.Component {
           metadata: temp_dataset,
           path,
         },
-      }).then((response) => {
-        this.setState({ dataset: temp_dataset, resource: {} });
-        alert("Resource has been removed sucessfully");
-      }).catch((error) => {
-        console.log(error);
-        alert("Error when removing resource!");
       })
+        .then((response) => {
+          this.setState({ dataset: temp_dataset, resource: {} });
+          alert("Resource has been removed sucessfully");
+        })
+        .catch((error) => {
+          console.log(error);
+          alert("Error when removing resource!");
+        });
     }
   };
 
@@ -255,14 +241,16 @@ export class DatasetEditor extends React.Component {
         metadata: this.state.dataset,
         description: this.state.dataset.description,
       },
-    }).then((response) => {
-      this.setState({ saveButtonText: "Save" });
-      alert("Uploaded Sucessfully");
-      this.setState({ currentStep: 0 });
-    }).catch((error) => {
-      console.log(error);
-      alert("Error on upload dataset!");
     })
+      .then((response) => {
+        this.setState({ saveButtonText: "Save" });
+        alert("Uploaded Sucessfully");
+        this.setState({ currentStep: 0 });
+      })
+      .catch((error) => {
+        console.log(error);
+        alert("Error on upload dataset!");
+      });
   };
 
   render() {
@@ -392,13 +380,13 @@ export class DatasetEditor extends React.Component {
                 Next
               </button>
             ) : (
-                <button disabled={true} className="btn">
-                  Next
-                </button>
-              )
+              <button disabled={true} className="btn">
+                Next
+              </button>
+            )
           ) : (
-              ""
-            )}
+            ""
+          )}
         </div>
       </div>
     );

--- a/src/App.js
+++ b/src/App.js
@@ -117,18 +117,27 @@ export class DatasetEditor extends React.Component {
 
   mapDatasetToFiscalFormat = (resource) => {
     const dataset = { ...this.state.dataset };
+    let model;
+    let schema;
+    if (dataset.resources.length > 0) {
+      //There's and existing resource, get schema and model
+      model = dataset.resources[0].model;
+      schema = dataset.resources[0].schema;
+      resource.schema = schema;
+      resource.model = model;
+    } else {
+      //first resource, generate schema and model
+      resource.schema.fields.forEach((f) => {
+        f.type = f.columnType;
+        delete f.columnType; //os-types requires type to be of rich type and will not accept the property colunmType
+      });
+      let fdp = new TypeProcessor().fieldsToModel(resource["schema"]["fields"]);
+      resource.schema.fields = Object.values(fdp.schema.fields);
+      resource.model = fdp.model;
+    }
+
     const resources = dataset.resources.map((res) => {
       if (res.hash == resource.hash) {
-        resource.schema.fields.forEach((f) => {
-          f.type = f.columnType;
-          delete f.columnType; //os-types requires type to be of rich type and will not accept the property colunmType
-        });
-        let fdp = new TypeProcessor().fieldsToModel(
-          resource["schema"]["fields"]
-        );
-
-        resource.schema.fields = Object.values(fdp.schema.fields);
-        resource.model = fdp.model;
         return resource;
       } else {
         return res;
@@ -199,11 +208,13 @@ export class DatasetEditor extends React.Component {
     if (status.success && !status.loading) {
       this.nextScreen();
     } else if (!status.success && status.error) {
-      const dataset = { ...this.state.dataset };
-      if ("resources" in dataset && dataset["resources"].length > 0) {
-        dataset.resources.pop();
-      }
-      this.setState({ dataset });
+      // const dataset = { ...this.state.dataset };
+      // if ("resources" in dataset && dataset["resources"].length > 0) {
+      //   dataset.resources.pop();
+      // }
+      // console.log("Here", dataset);
+
+      // this.setState({ dataset });
       this.prevScreen();
     }
 

--- a/src/App.js
+++ b/src/App.js
@@ -119,8 +119,8 @@ export class DatasetEditor extends React.Component {
     const dataset = { ...this.state.dataset };
     let model;
     let schema;
-    if (dataset.resources.length > 0) {
-      //There's and existing resource, get schema and model
+    if (dataset.resources.length > 1) {
+      //There's an existing resource, get schema and model
       model = dataset.resources[0].model;
       schema = dataset.resources[0].schema;
       resource.schema = schema;

--- a/src/App.js
+++ b/src/App.js
@@ -129,7 +129,7 @@ export class DatasetEditor extends React.Component {
       //first resource, generate schema and model
       resource.schema.fields.forEach((f) => {
         f.type = f.columnType;
-        delete f.columnType; //os-types requires type to be of rich type and will not accept the property colunmType
+        delete f.columnType; //os-types requires type to be of rich type and will not accept the property columnType
       });
       let fdp = new TypeProcessor().fieldsToModel(resource["schema"]["fields"]);
       resource.schema.fields = Object.values(fdp.schema.fields);

--- a/src/components/Upload/index.jsx
+++ b/src/components/Upload/index.jsx
@@ -5,7 +5,7 @@ import ProgressBar from "../ProgressBar";
 import { onFormatBytes, isValidURL } from "../../utils";
 import { Choose } from "datapub";
 import toArray from "stream-to-array";
-import "./Upload.css"
+import "./Upload.css";
 class Upload extends React.Component {
   constructor(props) {
     super(props);
@@ -26,33 +26,35 @@ class Upload extends React.Component {
       hashInProgress: false,
       hashLoaded: 0,
       uploadInProgress: false,
-      uploadedFileType: null
+      uploadedFileType: null,
     };
   }
 
-
   onChangeHandler = async (event) => {
     let { formattedSize, selectedFile } = this.state;
-    let path = ""
+    let path = "";
     if (event.target.type == "file" && event.target.files.length > 0) {
       selectedFile = event.target.files[0];
-      path = `data/${selectedFile.name}`
-      this.setState({ uploadedFileType: "file" })
+      path = `data/${selectedFile.name}`;
+      this.setState({ uploadedFileType: "file" });
     } else {
-      selectedFile = event.target.value
+      selectedFile = event.target.value;
       if (!isValidURL(selectedFile)) {
-        this.setErrorState("Invalid URL! Please ensure entered URL is correct")
-        return
+        this.setErrorState("Invalid URL! Please ensure entered URL is correct");
+        return;
       }
-      this.setState({ uploadedFileType: "url" })
-      path = selectedFile
+      this.setState({ uploadedFileType: "url" });
+      path = selectedFile;
     }
 
-    const { validFile, errorMsg, file } = await this.validFileSelected(selectedFile, event.target.type)
+    const { validFile, errorMsg, file } = await this.validFileSelected(
+      selectedFile,
+      event.target.type
+    );
 
     if (!validFile) {
-      this.setErrorState(errorMsg)
-      return
+      this.setErrorState(errorMsg);
+      return;
     }
 
     formattedSize = onFormatBytes(file.size);
@@ -62,24 +64,25 @@ class Upload extends React.Component {
       self.onHashProgress(progress);
     });
 
-
-    Object.assign(file.descriptor, { hash })
+    Object.assign(file.descriptor, { hash });
 
     //check if file has the same schema
     if (!this.hasSameSchema(file._descriptor)) {
-      this.setErrorState("Schema of uploaded resource does not match existing one!")
-      return
+      this.setErrorState(
+        "Schema of uploaded resource does not match existing one!"
+      );
+      return;
     }
 
     //check if file already exists in resource
     if (this.hasSameHash(file._descriptor)) {
-      this.setErrorState("Possible duplicate, the resource already exists!")
-      return
+      this.setErrorState("Possible duplicate, the resource already exists!");
+      return;
     }
 
     //get sample
     let sample_stream = await file.rows({ size: 460 });
-    let sample = await toArray(sample_stream)
+    let sample = (await toArray(sample_stream)).slice(0, 30);
     //get column names for table
     const column_names = sample[0]; //first row is the column names
     const tablePreviewColumns = column_names.map((item) => {
@@ -91,7 +94,7 @@ class Upload extends React.Component {
 
     //prepare sample for use in table preview component
     let tablePreviewSample = [];
-    sample.slice(1,11).forEach((item) => {
+    sample.slice(1, 11).forEach((item) => {
       let temp_obj = {};
       item.forEach((field, i) => {
         temp_obj[column_names[i]] = field;
@@ -99,9 +102,13 @@ class Upload extends React.Component {
       tablePreviewSample.push(temp_obj);
     });
 
-   
     this.props.metadataHandler(
-      Object.assign(file.descriptor, { sample, tablePreviewSample, tablePreviewColumns, path })
+      Object.assign(file.descriptor, {
+        sample,
+        tablePreviewSample,
+        tablePreviewColumns,
+        path,
+      })
     );
 
     this.setState({
@@ -114,75 +121,83 @@ class Upload extends React.Component {
     });
 
     await this.onClickHandler();
-
-
   };
 
   validFileSelected = (selectedFile, fileType) => {
     return new Promise(async (resolve, reject) => {
       if (fileType == "url") {
-        const fileExt = selectedFile.split(".").pop()
+        const fileExt = selectedFile.split(".").pop();
         if (fileExt != "csv") {
-          resolve({ validFile: false, errorMsg: "File Type not supported! Please upload a CSV file", file: undefined })
-          return
+          resolve({
+            validFile: false,
+            errorMsg: "File Type not supported! Please upload a CSV file",
+            file: undefined,
+          });
+          return;
         }
-        fetch(selectedFile).then(async (resp) => {
-          let file;
-          try {
-            file = data.open(selectedFile)
-            await file.addSchema()
-            resolve({
-              validFile: true,
-              errorMsg: "", file
-            })
-          } catch (error) {
+        fetch(selectedFile)
+          .then(async (resp) => {
+            let file;
+            try {
+              file = data.open(selectedFile);
+              await file.addSchema();
+              resolve({
+                validFile: true,
+                errorMsg: "",
+                file,
+              });
+            } catch (error) {
+              console.log(error);
+              resolve({
+                validFile: false,
+                errorMsg: "An error occured when trying to load the file!",
+                file,
+              });
+            }
+          })
+          .catch((error) => {
             console.log(error);
             resolve({
               validFile: false,
-              errorMsg: "An error occured when trying to load the file!", file
-            })
-          }
-        }).catch((error) => {
-          console.log(error);
-          resolve({
-            validFile: false,
-            errorMsg: "An error occured when trying to load the file!"
-          })
-        })
+              errorMsg: "An error occured when trying to load the file!",
+            });
+          });
       } else {
-        const fileExt = selectedFile.type.split("/").pop()
+        const fileExt = selectedFile.type.split("/").pop();
 
         if (fileExt != "csv") {
           resolve({
             validFile: false,
             errorMsg: "File Type not supported! Please upload a CSV file",
-            file: {}
-          })
-          return
+            file: {},
+          });
+          return;
         }
         if (selectedFile.size == 0) {
           resolve({
             validFile: false,
-            errorMsg: "CSV file is empty! Please upload a CSV file with contents",
-            file: {}
-          })
-          return
+            errorMsg:
+              "CSV file is empty! Please upload a CSV file with contents",
+            file: {},
+          });
+          return;
         }
 
         try {
-          const file = data.open(selectedFile)
-          await file.addSchema()
-          resolve({ validFile: true, errorMsg: "", file })
+          const file = data.open(selectedFile);
+          await file.addSchema();
+          resolve({ validFile: true, errorMsg: "", file });
         } catch (error) {
           console.log(error);
           resolve({
             validFile: false,
-            errorMsg: "An error occured when trying to load the file!", file: {}
-          })
+            errorMsg: "An error occured when trying to load the file!",
+            file: {},
+          });
         }
       }
-    })
-  }
+    });
+  };
 
   setErrorState(errorMsg) {
     this.setState({ error: true, loading: false });
@@ -190,7 +205,7 @@ class Upload extends React.Component {
       loading: false,
       success: false,
       error: true,
-      errorMsg
+      errorMsg,
     });
   }
 
@@ -203,11 +218,13 @@ class Upload extends React.Component {
   };
 
   onUploadProgress = (progressEvent) => {
-    const loaded = Number(((progressEvent.loaded / progressEvent.total) * 100).toFixed())
+    const loaded = Number(
+      ((progressEvent.loaded / progressEvent.total) * 100).toFixed()
+    );
     this.onTimeRemaining(progressEvent.loaded);
     this.setState({
       loaded,
-      uploadInProgress: true
+      uploadInProgress: true,
     });
   };
 
@@ -223,30 +240,36 @@ class Upload extends React.Component {
   };
 
   hasSameSchema = (resource) => {
-    if (Object.keys(this.state.dataset).includes("resources") && this.state.dataset.resources.length > 0) {
-
+    if (
+      Object.keys(this.state.dataset).includes("resources") &&
+      this.state.dataset.resources.length > 0
+    ) {
       const newFields = resource.schema.fields.map((field) => {
-        return field.name
-      })
-      const oldFields = this.state.dataset.resources[0].schema.fields.map((field) => {
-        return field.name
-      })
+        return field.name;
+      });
+      const oldFields = this.state.dataset.resources[0].schema.fields.map(
+        (field) => {
+          return field.name;
+        }
+      );
       return JSON.stringify(newFields) === JSON.stringify(oldFields);
     } else {
-      return true
+      return true;
     }
-  }
+  };
 
   hasSameHash(newResource) {
-    if (Object.keys(this.state.dataset).includes("resources")
-      && this.state.dataset.resources.length > 0) {
-      const { resources } = this.state.dataset
+    if (
+      Object.keys(this.state.dataset).includes("resources") &&
+      this.state.dataset.resources.length > 0
+    ) {
+      const { resources } = this.state.dataset;
       const sameHashes = resources.map((resource) => {
-        return resource.hash == newResource.hash
-      })
-      return sameHashes.includes(true)
+        return resource.hash == newResource.hash;
+      });
+      return sameHashes.includes(true);
     } else {
-      return false
+      return false;
     }
   }
 
@@ -282,7 +305,12 @@ class Upload extends React.Component {
       });
 
       client
-        .upload(resource, organizationId, this.state.datasetId, this.onUploadProgress)
+        .upload(
+          resource,
+          organizationId,
+          this.state.datasetId,
+          this.onUploadProgress
+        )
         .then((response) => {
           this.setState({
             success: true,
@@ -303,12 +331,11 @@ class Upload extends React.Component {
             loading: false,
             success: false,
             error: true,
-            errorMsg: `Upload failed with error: ${error.message}`
+            errorMsg: `Upload failed with error: ${error.message}`,
           });
         });
     }
-  }
-
+  };
 
   render() {
     const {
@@ -322,8 +349,7 @@ class Upload extends React.Component {
       uploadInProgress,
     } = this.state;
     return (
-
-      <div >
+      <div>
         <Choose
           onChangeHandler={this.onChangeHandler}
           onChangeUrl={this.onChangeHandler}
@@ -343,7 +369,6 @@ class Upload extends React.Component {
                 circleTwoStroke={"#7ea9e1"}
               />
             </>
-
           </div>
         )}
         {uploadInProgress && (
@@ -361,7 +386,7 @@ class Upload extends React.Component {
                 strokeWidth={5}
                 circleOneStroke="#d9edfe"
                 circleTwoStroke={"#7ea9e1"}
-              // timeRemaining={timeRemaining}
+                // timeRemaining={timeRemaining}
               />
               <h2>
                 {success &&
@@ -372,7 +397,6 @@ class Upload extends React.Component {
                 {error && "Upload failed"}
               </h2>
             </>
-
           </div>
         )}
       </div>

--- a/src/components/Upload/index.jsx
+++ b/src/components/Upload/index.jsx
@@ -191,7 +191,7 @@ class Upload extends React.Component {
           console.log(error);
           resolve({
             validFile: false,
-            errorMsg: "An error occured when trying to load the file!",
+            errorMsg: "An error occurred when trying to load the file!",
             file: {},
           });
         }

--- a/src/components/Upload/index.jsx
+++ b/src/components/Upload/index.jsx
@@ -150,7 +150,7 @@ class Upload extends React.Component {
               console.log(error);
               resolve({
                 validFile: false,
-                errorMsg: "An error occured when trying to load the file!",
+                errorMsg: "An error occurred when trying to load the file!",
                 file,
               });
             }

